### ISSUE TITLE
🐛 BUG: D1 Maximum call stack size exceeded

### ISSUE THREAD

**Issue by o0th** [cloudflare/wrangler2#2509]
### Which Cloudflare product(s) does this pertain to?

D1

### What version of `Wrangler` are you using?

2.6.2

### What operating system are you using?

Linux

### Describe the Bug

I was trying to execute this migration script on D1

```sql
-- Migration number: 0002 	 2022-11-24T10:16:43.132Z

drop table if exists `countries_markets`;

create table if not exists `countries_markets` (
  `countryID` integer not null,
  `marketID` integer not null,
  foreign key (`countryID`) references `countries`(`countryID`),
  foreign key (`marketID`) references `markets`(`marketID`)
);

insert into `countries_markets` (`countryID`, `marketID`)
  select
    (select `countryID` from `countries` where `countryCode` = 'af') as `countryID`,
    (select `marketID` from `markets` where `marketCode` = 'we') as `marketID`
  union all select
    (select `countryID` from `countries` where `countryCode` = 'ax') as `countryID`,
    (select `marketID` from `markets` where `marketCode` = 'fi') as `marketID`

-- continue --
```

and got this error

```
Error 9000: exception caught in D1 core. Maximum call stack size exceeded RangeError: Maximum call stack size exceeded│
     at wasm://wasm/00f3ee4a:wasm-function[3830]:0x24c422
     at wasm://wasm/00f3ee4a:wasm-function[3830]:0x254ab0
     at wasm://wasm/00f3ee4a:wasm-function[3830]:0x254ab0
     at wasm://wasm/00f3ee4a:wasm-function[3830]:0x254ab0
     at wasm://wasm/00f3ee4a:wasm-function[3830]:0x254ab0
     at wasm://wasm/00f3ee4a:wasm-function[3830]:0x254ab0
     at wasm://wasm/00f3ee4a:wasm-function[3830]:0x254ab0
     at wasm://wasm/00f3ee4a:wasm-function[3830]:0x254ab0
     at wasm://wasm/00f3ee4a:wasm-function[3830]:0x254ab0
     at wasm://wasm/00f3ee4a:wasm-function[3830]:0x254ab0 [code: 7500]
```

Using `--local` everything works fine

**Comment by o0th** [comment]
I think the problem is with `union all select` since everything works fine with multiple insert statements.

```sql
-- Migration number: 0002 	 2022-11-24T10:16:43.132Z

drop table if exists `countries_markets`;

create table if not exists `countries_markets` (
  `countryID` integer not null,
  `marketID` integer not null,
  foreign key (`countryID`) references `countries`(`countryID`),
  foreign key (`marketID`) references `markets`(`marketID`)
);

insert into `countries_markets` (`countryID`, `marketID`)
  select    
    (select `countryID` from `countries` where `countryCode` = 'af') as `countryID`,
    (select `marketID` from `markets` where `marketCode` = 'we') as `marketID`;

insert into `countries_markets` (`countryID`, `marketID`)
  select    
    (select `countryID` from `countries` where `countryCode` = 'ax') as `countryID`,
    (select `marketID` from `markets` where `marketCode` = 'fi') as `marketID`;

-- continue --
```

**Comment by o0th** [comment]
This issue seems fixed with version `>= 2.7.0`
